packages/dds/tree/src/simple-tree/discrepancies.ts

```diff
@@ -158,75 +218,113 @@ function getNodeSchemaType(nodeSchema: TreeNodeStoredSchema): SchemaFactoryNodeK
  * It is up to the caller to determine whether such discrepancies matter.
  */
 export function* getAllowedContentDiscrepancies(
-	view: TreeStoredSchema,
+	view: FieldSchema,
 	stored: TreeStoredSchema,
 ): Iterable<Discrepancy> {
 	// check root schema discrepancies
-	yield* getFieldDiscrepancies(
-		view.rootFieldSchema,
-		stored.rootFieldSchema,
-		undefined,
-		undefined,
-	);
+	yield* getFieldDiscrepancies(view, stored.rootFieldSchema, undefined, undefined);
 
-	for (const result of compareMaps(view.nodeSchema, stored.nodeSchema)) {
-		switch (result.type) {
-			case "aExtra": {
-				const viewType = getNodeSchemaType(result.value);
-				yield {
-					identifier: result.key,
-					mismatch: "nodeKind",
-					view: viewType,
-					stored: undefined,
-				};
-				break;
-			}
-			case "bExtra": {
-				const storedType = getNodeSchemaType(result.value);
-				yield {
-					identifier: result.key,
-					mismatch: "nodeKind",
-					view: undefined,
-					stored: storedType,
-				};
-				break;
-			}
-			case "both": {
-				yield* getNodeDiscrepancies(result.key, result.valueA, result.valueB);
-				break;
-			}
-			default:
-				break;
+	const viewNodeSchema = new Map<TreeNodeSchemaIdentifier, TreeNodeSchema>();
+
+	walkFieldSchema(view, {
+		node: (schema) => {
+			const identifier: TreeNodeSchemaIdentifier = brand(schema.identifier);
+
+			debugAssert(() => !viewNodeSchema.has(identifier));
+			viewNodeSchema.set(identifier, schema);
+		},
+	});
+
+	for (const [identifier, viewSchema] of viewNodeSchema) {
+		const storedSchema = stored.nodeSchema.get(identifier);
+
+		// if the view schema has a node that's not in the stored schema
+		if (storedSchema === undefined) {
+			const viewType = getViewNodeSchemaType(viewSchema);
+			// TODO does it make sense to have this mismatch when there will also be an allowedTypes mismatch?
```

Comment:
I added a comment about this where the discrepancies are handled. If you include that note, maybe reference it from here or have that one reference this one?

I think we can keep it for now, but might want to remove it later